src/lib.rs

```diff
@@ -910,14 +907,15 @@ impl Unicorn {
     ///
     /// `hook` is the value returned by either `add_code_hook` or `add_mem_hook`.
     pub fn remove_hook(&mut self, hook: uc_hook) -> Result<()> {
-        let err = unsafe { uc_hook_del(self.handle, hook) } as Error;
+        let err = unsafe { uc_hook_del(self.handle, hook) };
         // Check in all maps to find which one has the hook.
-        self.code_callbacks.remove(&hook);
-        self.intr_callbacks.remove(&hook);
-        self.mem_callbacks.remove(&hook);
-        self.insn_in_callbacks.remove(&hook);
-        self.insn_out_callbacks.remove(&hook);
-        self.insn_sys_callbacks.remove(&hook);
+        macro_rules! ignore { () => { |_| () } };
+        self.code_callbacks.remove(&hook).map(ignore!())
+            .or_else(|| self.intr_callbacks.remove(&hook).map(ignore!()))
+            .or_else(|| self.mem_callbacks.remove(&hook).map(ignore!()))
+            .or_else(|| self.insn_in_callbacks.remove(&hook).map(ignore!()))
+            .or_else(|| self.insn_out_callbacks.remove(&hook).map(ignore!()))
+            .or_else(|| self.insn_sys_callbacks.remove(&hook).map(ignore!()));
```

Comment:
I think this could be simplified to

```rust
     let _ = self.code_callbacks.remove(&hook)
            .or_else(|| self.intr_callbacks.remove(&hook)
            .or_else(|| self.mem_callbacks.remove(&hook)
            .or_else(|| self.insn_in_callbacks.remove(&hook)
            .or_else(|| self.insn_out_callbacks.remove(&hook)
            .or_else(|| self.insn_sys_callbacks.remove(&hook);
```

Reply:
Hi @ekse and thanks for the response.

But this code will not compile since the callbacks have different types, each `remove` returns an `Option<T>` where `T` is unique for each callback. That's the reason why I've used `ignore!()` to map all `T -> ()`.

For boxed slice v.s vector, yes, `boxed slice` is not as popular as vector (though it's trivial to convert between them). So if you don't want it then I will remove it.

Reply:
Oh you're right, let's go with your approach then. This will be ready to merge after the change back to Vec.